sklearn/preprocessing/tests/test_target.py

```diff
@@ -0,0 +1,263 @@
+    sample_weight = np.ones((y.shape[0],))
+    regr = TransformedTargetRegressor(regressor=Lasso(),
+                                      transformer=StandardScaler())
+    assert_raises_regex(TypeError, "fit\(\) got an unexpected keyword argument"
+                        " 'sample_weight'", regr.fit, X, y,
+                        sample_weight=sample_weight)
+    # func is given but inverse_func is not
+    regr = TransformedTargetRegressor(func=np.exp)
+    assert_raises_regex(ValueError, "When 'func' is not None, 'inverse_func'"
+                        " cannot be None.", regr.fit, X, y)
+
+
+def test_transform_target_regressor_invertible():
+    X, y = friedman
+    regr = TransformedTargetRegressor(regressor=LinearRegression(),
+                                      func=np.sqrt, inverse_func=np.log,
+                                      check_inverse=True)
+    assert_warns_message(UserWarning, "The provided functions or transformer"
+                         " are not strictly inverse of each other.",
+                         regr.fit, X, y)
+    regr = TransformedTargetRegressor(regressor=LinearRegression(),
+                                      func=np.sqrt, inverse_func=np.log,
+                                      check_inverse=False)
+    # the transformer/functions are not checked to be invertible the fitting
+    # should pass
+    assert_no_warnings(regr.fit, X, y)
+
+
+def _check_standard_scaled(y, y_pred):
+    y_mean = np.mean(y, axis=0)
+    y_std = np.std(y, axis=0)
+    assert_allclose((y - y_mean) / y_std, y_pred)
+
+
+def _check_shifted_by_one(y, y_pred):
+    assert_allclose(y + 1, y_pred)
+
+
+def test_transform_target_regressor_functions():
+    X, y = friedman
+    regr = TransformedTargetRegressor(regressor=LinearRegression(),
+                                      func=np.log, inverse_func=np.exp)
+    y_pred = regr.fit(X, y).predict(X)
+    # check the transformer output
+    y_tran = regr.transformer_.transform(y.reshape(-1, 1)).squeeze()
+    assert_allclose(np.log(y), y_tran)
+    assert_allclose(y, regr.transformer_.inverse_transform(
+        y_tran.reshape(-1, 1)).squeeze())
+    assert_equal(y.shape, y_pred.shape)
+    assert_allclose(y_pred, regr.inverse_func(regr.regressor_.predict(X)))
+    # check the regressor output
+    lr = LinearRegression().fit(X, regr.func(y))
+    assert_allclose(regr.regressor_.coef_.ravel(), lr.coef_.ravel())
+
+
+def test_transform_target_regressor_functions_multioutput():
+    X = friedman[0]
+    y = np.vstack((friedman[1], friedman[1] ** 2 + 1)).T
+    regr = TransformedTargetRegressor(regressor=LinearRegression(),
+                                      func=np.log, inverse_func=np.exp)
+    y_pred = regr.fit(X, y).predict(X)
+    # check the transformer output
+    y_tran = regr.transformer_.transform(y)
+    assert_allclose(np.log(y), y_tran)
+    assert_allclose(y, regr.transformer_.inverse_transform(y_tran))
+    assert_equal(y.shape, y_pred.shape)
```

Comment:
with pytest, we can just use a bare `assert y.shape == y_pred.shape`, which I find much more legible.